src/views/DPI/components/DpiPriceChanges.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { ReactNode } from 'react'
+import styled from 'styled-components'
+import numeral from 'numeral'
+
+import InfoSection from './InfoSection'
+
+import useDpiTokenMarketData from 'hooks/useDpiTokenMarketData'
+
+const DpiPriceChanges: React.FC = () => {
+  const { prices } = useDpiTokenMarketData()
+  const [startTime] = prices?.[0] || [0]
+  const [latestTime, latestPrice] = prices?.[prices.length - 1] || [0, 0]
+  // Coingecko API returns each item as hourly data if < 90 days and daily data > 90 days
+  // So if time between start and end time < 90 days, 1 day of data is 24 array items.
+  const dataCadenceFor24Hours = latestTime - startTime < 7776000000 ? 24 : 1
+  const priceChangeIntervals = [
+    ['1 Day', 1],
+    ['1 Month', 30],
+    ['3 Months', 90],
+    ['1 Year', 365],
+  ]
+
+  const calculatePriceDiffs = (daysOfComparison: number) => {
```

Comment:
This function is doing two things: calculating the price difference & returning a JSX component.
Maybe separating the price change calculation into a util would help make it a bit easier to read.
Also just renaming the function to `renderAssetPriceChange` could help.